### PR TITLE
[IMP] fieldservice_stock: SRO to SR Tree

### PR DESCRIPTION
**Pull request by osi-scampbell**
Add SRO's to SR Tree on Field Service Orders

### CODE REVIEW

File: fieldservice_stock/models/stock_request.py [hunk 1]
```diff
@@ -41,17 +41,20 @@ def create(self, vals):
...
             if not isinstance(vals['expected_date'], str):
-                val_date = datetime.strftime(vals['expected_date'],
-                                             '%Y-%m-%d %H:%M:%S')
-
+                val_date = datetime.strftime(vals['expected_date'], '%Y-%m-%d %H:%M:%S')
             val_date = datetime.strptime(val_date, '%Y-%m-%d %H:%M:%S')
+            picking_type_id = self.env['stock.picking.type'].search([
+                    ('code', '=', 'stock_request_order'),
+                    ('warehouse_id', '=', vals['warehouse_id'])],
+                    limit=1).id
```
Comment (max3903) [inline fieldservice_stock/models/stock_request.py]:
```suggestion
                    limit=1)
```

File: fieldservice_stock/models/stock_request.py [hunk 2]
```diff
@@ -41,17 +41,20 @@ def create(self, vals):
...
             date_window_after = val_date - timedelta(hours=1)
-
             order = self.env['stock.request.order'].search([
                 ('fsm_order_id', '=', vals['fsm_order_id']),
+                ('warehouse_id', '=', vals['warehouse_id']),
+                ('picking_type_id', '=', picking_type_id),
```
Comment (max3903) [inline fieldservice_stock/models/stock_request.py]:
```suggestion
                ('picking_type_id', '=', picking_type_id.id),
```

File: fieldservice_stock/models/stock_request.py [hunk 3]
```diff
@@ -61,6 +64,10 @@ def create(self, vals):
                 vals['order_id'] = order.id
             else:
                 values = self.prepare_order_values(vals)
+                values.update({
+                    'picking_type_id':picking_type_id,
```
Comment (max3903) [inline fieldservice_stock/models/stock_request.py]:
```suggestion
                    'picking_type_id':picking_type_id.id,
```

**Review by max3903 — CHANGES_REQUESTED** [review]

**Review by patrickrwilson — APPROVED** [review]
Please fix LINT errors, then it looks good to me.

**Review by bodedra — APPROVED** [review]
LGTM :+1:

**Review by max3903 — APPROVED** [review]

**Pull request merged**